benchmark_runner/common/prometheus_snapshot/prometheus_snapshot.py

```diff
@@ -0,0 +1,83 @@
+
+import datetime
+import time
+import string
+from benchmark_runner.common.prometheus_snapshot.prometheus_snapshot_exceptions import PrometheusSnapshotError, PrometheusSnapshotAlreadyStarted, PrometheusSnapshotNotStarted, PrometheusSnapshotAlreadyRetrieved
+
+from typeguard import typechecked
+from benchmark_runner.common.logger.logger_time_stamp import logger_time_stamp, logger
+from benchmark_runner.common.oc.oc import OC
+import os
+import re
+
+prometheus_snapshot_datetime_format = '%Y_%m_%dT%H_%M_%S%z'
+
+class PrometheusSnapshot:
+    """
+    Extract a Prometheus snapshot into a specified directory.
+    The directory is assumed to already exist
+    """
+    def __init__(self, oc: OC, log_path: str, verbose: bool=False):
+        self.__oc = oc
+        self.__log_path = log_path
+        self.__start_timestamp = None
+        self.__end_timestamp = None
+        self.__verbose = verbose
+        self.__pod_name = 'prometheus-k8s-0'
+        self.__namespace = 'openshift-monitoring'
+        self.__container = 'prometheus'
+        self.__format = prometheus_snapshot_datetime_format
+
+    @typechecked
+    def __verbose_log(self, message: str):
+        if self.__verbose:
+            logger.info(message)
+
+    @typechecked
+    def __exec(self, command: str):
+        return self.__oc.exec(pod_name=self.__pod_name, namespace=self.__namespace, container=self.__container, command=command)
+
+    def __prom_timestamp(self):
+        valid_timestamp = False
+        pattern = r'^[0-9]{4}'
+        retries = 12
+        while not valid_timestamp:
+            timestamp = self.__exec(f"date '+{self.__format}'")
```

Comment:
Why you are using + and f
Is it working ?

Reply:
Please all class and methods descriptions

Reply:
This amounts to

```
date +$format
```

This isn't string concatenation.

Reply:
Why do you need to amount it there and not here ?

Reply:
I don't understand the question?

Reply:
This is running the `date` command on the prometheus pod.